frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -1414,6 +1414,15 @@ const IR::Node* TypeInference::postorder(IR::P4ValueSet* decl) {
             if (errs)
                 return nullptr;
         }
+        if (!canon->is<IR::Type_Newtype>() &&
```

Comment:
This check seems a little awkward, what is the common characteristic of these types? That they have a fixed width?

Reply:
Headers may have fixed widths too, but are not allowed.